src/main/java/soconnect/commons/util/StringUtil.java

```diff
@@ -55,22 +54,54 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
     public static boolean containsKeywordsIgnoreCase(String sentence, String keywords) {
         requireAllNonNull(sentence, keywords);
 
-        List<String> preppedKeywords = Arrays.asList(keywords.toUpperCase().trim().split("\\s+"));
+        String preppedWord = keywords.trim();
+        checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
+        List<String> preppedKeywords = Arrays.asList(preppedWord.toUpperCase().trim().split("\\s+"));
         checkArgument(preppedKeywords.size() > 0, "Word parameter cannot be empty");
 
         List<String> wordsInPreppedSentence = Arrays.asList(sentence.toUpperCase().split("\\s+"));
 
         return wordsInPreppedSentence.containsAll(preppedKeywords);
     }
 
+    /**
+     * Returns true if the {@code sentence} contains some of the {@code keywords}.
+     *   Ignores case and order, but full keywords match is not required.
+     *   In the event of not match, decrease the keywords size by one-forth.
```

Comment:
```suggestion
     *   In the event of not match, decrease the keywords size by one-fourth.
```
Also why one-fourth? Seems like a magic-literal without accompanying explainations. [Magic Numbers](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)

Reply:
It's just my consideration of how precise the relevant search should be. The current algo is like this:
1. It will match all occurrences of characters in keywords to information. If it contains all, then we will just return the search result. This benefit user that misspell words like `misspell` to `mispell`, `john` to `jhon`, `occurrences` to `ocurrences`, these kind of things which actually occur when I typed this😂.
2. If first step does not generate search result, we then decrease the number of characters in keywords need to be matched. Based on some trial and errors and surfing [articles](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2137159/) in Google, I roughly conclude that 25% reduction is kind of a sweet spot to allow for precise relevant result. Number higher than that will likely to just return irrelevant contacts and number smaller than that might fall into step 1.

But if you all agree with this algo, I think I can paste this explanation for the documentation :D

Reply:
Smart 👍 I think this algorithm can be a defining portion of your search feature implementation in the DG. Can considering including a condensed/simplified explaination in the Javadocs here to avoid receiving 'Magic Number' bug reports 😄